library/packages/src/lib/packages/repository_product.rb

```diff
@@ -0,0 +1,48 @@
+module Packages
+  # Represent a product which is present in a repository. At this
+  # time this class is responsible for finding out whether two
+  # products instances are the same (for example, coming from different
+  # repositories).
+  class RepositoryProduct
```

Comment:
NP: Maybe simple `Product` could be enough as it already is in the `Packages` name space, there should not be any conflict. IMHO `Packages::Product` looks better.